webhooks/mutatepod_parts.go

```diff
@@ -105,33 +139,65 @@ func gcloudSetupContainer(
 
 // VolumeMounts
 var (
-	volumeMountsToAddOrReplace = []corev1.VolumeMount{k8sSATokenVolumeMount, gcloudConfigVolumeMount}
-
+	externalCredConfigVolumeMount = corev1.VolumeMount{
+		Name:      DirectInjectedExternalVolumeName,
+		MountPath: DirectInjectedExternalMountPath,
+		ReadOnly:  true,
+	}
 	k8sSATokenVolumeMount = corev1.VolumeMount{
 		Name:      K8sSATokenVolumeName,
 		MountPath: K8sSATokenMountPath,
 		ReadOnly:  true,
 	}
 	gcloudConfigVolumeMount = corev1.VolumeMount{
 		Name:      GCloudConfigVolumeName,
-		MountPath: GCloudConifgMountPath,
+		MountPath: GCloudConfigMountPath,
 	}
 )
 
+func volumeMountsToAddOrReplace(mode InjectionMode) []corev1.VolumeMount {
+	volMounts := []corev1.VolumeMount{k8sSATokenVolumeMount}
+
+	if mode == DirectMode {
+		volMounts = append(volMounts, externalCredConfigVolumeMount)
+	} else {
+		volMounts = append(volMounts, gcloudConfigVolumeMount)
+	}
+
+	return volMounts
+}
+
 // EnvVars
 var (
-	envVarsToAddOrReplace = []corev1.EnvVar{googleAppCredentialsEnvVar, cloudSDKConfigEnvVar}
-
 	googleAppCredentialsEnvVar = corev1.EnvVar{
 		Name:  "GOOGLE_APPLICATION_CREDENTIALS",
-		Value: filepath.Join(GCloudConifgMountPath, "federation.json"),
+		Value: filepath.Join(GCloudConfigMountPath, ExternalCredConfigFilename),
 	}
```

Comment:
nit: This variable can be moved to `mutatepod_parts_test.go` now.

Reply:
Thanks for pointing this out. I didn't know about `%w` for errors.

Reply:
I had a look but I'm not sure what you mean, are you able to provide some more detail please?

Reply:
Sorry, I added the comment to the wrong line. I will make code suggestions.

Reply:
I meant, now that `googleAppCredentialsEnvVar` is referred only in `webhooks/mutatepod_parts_test.go`, the variable declaration can be moved to `webhooks/muatepod_parts_test.go`. I will make code suggestions.

Reply:
Ah yes I understand now, I thought you meant the `ExternalCredConfigFilename` variable because that's the diff line highlighted in the code snippet above, but I see you meant `googleAppCredentialsEnvVar`.

Reply:
Right. Sorry for my lack of explanation.